File: Backend/src/main/java/com/cts/entity/PassengerProfile.java
package com.cts.entity;

import com.cts.enumeration.IdProofType;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "passenger_profiles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PassengerProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long profileId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private User customer;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Integer age;

    @Column(nullable = false)
    private String gender;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IdProofType idProofType;

    @Column(nullable = false)
    private String idProofNumber;
}

File: Frontend/src/app/shared/models/passenger-profile.ts
export interface PassengerProfile {
  profileId?: number; 
  name: string;
  age: number;
  gender: 'Male' | 'Female';
  idProofType: 'PAN' | 'AADHAAR' | 'PASSPORT' | 'DRIVING_LICENSE';
  idProofNumber: string;
}

export interface PassengerSnapshot {
  name: string;
  age: number;
  gender: string;
  idProofType: string;
  idProofNumber: string;
}